### ISSUE TITLE
[BUG] Typechecking UNCHANGED tuple of variables

### ISSUE THREAD

**Issue by Isaac-DeFrain** [informalsystems/apalache#744]
## Description

Snowcat requests an annotation for the UNCHANGED operator when applied to a tuple of variables.

## Input specification

```
---- MODULE Example ----

EXTENDS Naturals

VARIABLE
    \* @type: Int;
    x,
    \* @type: Int;
    y,
    \* @type: Int;
    z

\* @type: <<Int, Int, Int>>;
vars == <<x, y, z>>

Incr == \E n \in 1..10 :
    /\ x' = x + n
    /\ UNCHANGED <<y, z>>

Init ==
    /\ x = 0
    /\ y = 0
    /\ z = 0

Next == Incr

Spec == Init /\ [][Next]_vars

========================
```

## The command line parameters used to run the tool

$ apalache typecheck Example.tla

## Expected behavior

Snowcat should not need an annotation for UNCHANGED in this context.

## Log files

```
PASS #1: TypeChecker                                              I@22:21:25.112
 > Running Snowcat .::.                                           I@22:21:25.112
[Example.tla:18:18-18:25]: Need annotation. Arguments match 2 operator signatures: ((i, j) => <<i, j>>) and ((i, i) => Seq(i)) E@22:21:25.211
[Example.tla:16:1-18:25]: Error when computing the type of Incr   E@22:21:25.212
 > Snowcat asks you to fix the types. Meow.                       I@22:21:25.212
Type checker [FAILED]                                             I@22:21:25.212
```

## System information

- Apalache version [`apalache-mc version`]: 0.11.1-SNAPSHOT build v0.11.0-17-gf55663c
- OS [e.g. Ubuntu Linux or Mac OS, and the current version]: Debian 10
- JDK version [e.g. OpenJDK 0.8.0]: openjdk 11.0.9.1 2020-11-04

## Additional context

The following, equivalent, spec typechecks successfully:

```
---- MODULE Example ----

EXTENDS Naturals

VARIABLE
    \* @type: Int;
    x,
    \* @type: Int;
    y,
    \* @type: Int;
    z

\* @type: <<Int, Int, Int>>;
vars == <<x, y, z>>

Incr == \E n \in 1..10 :
    /\ x' = x + n
    /\ UNCHANGED y
    /\ UNCHANGED z

Init ==
    /\ x = 0
    /\ y = 0
    /\ z = 0

Next == Incr

Spec == Init /\ [][Next]_vars

========================
```

**Comment by konnov** [comment]
We have fixed it in 0.15.x. Can you try it with 0.15.3?

**Comment by Isaac-DeFrain** [comment]
This bug was reported for 0.11.1-SNAPSHOT build v0.11.0-17-gf55663c, it is not present in 0.15.3-SNAPSHOT build v0.15.1-33-g80cf914.
